replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,405 @@
+
+  private volatile boolean processing = false;
+
+  private volatile boolean running = true;
+
+  private final NodeAdapter local;
+
+  private final Queue queue;
+
+  private final SiteManager siteManager;
+
+  private final NodeAdapters nodeAdapters;
+
+  /**
+   * Creates a new worker which will monitor the provided queue and perform operations between the
+   * provided local and destination sites.
+   *
+   * <p>Does not begin monitoring of the queue until {@link #run()} is called.
+   *
+   * @param queue the queue to monitor tasks for
+   * @param local the local site's node adapter
+   * @param siteManager site manager for accessing sites
+   * @param nodeAdapters node adapter factory
+   */
+  Worker(Queue queue, NodeAdapter local, SiteManager siteManager, NodeAdapters nodeAdapters) {
+    this.queue = queue;
+    this.local = local;
+    this.siteManager = siteManager;
+    this.nodeAdapters = nodeAdapters;
+  }
+
+  /**
+   * Begin monitoring a queue for tasks. The handler of this worker is responsible for halting this
+   * thread by calling interrupt on this worker's thread, finishing the transfer of the current task
+   * if necessary.
+   */
+  @Override
+  public void run() {
+    try {
+      while (running) {
+        if (doRun()) {
+          break;
+        }
+      }
+    } finally {
+      try {
+        local.close();
+      } catch (IOException ie) {
+        LOGGER.debug("Failed to close adapter", ie);
+      }
+    }
+  }
+
+  /** @return {@code true} if should exit, otherwise {@code false} */
+  @VisibleForTesting
+  boolean doRun() {
+    Task task = null;
+    try {
+      task = queue.take();
+      this.processing = true;
+      if (supported(task)) {
+        this.handle(task);
+      } else {
+        task.unlock();
+      }
+      this.processing = false;
+    } catch (InterruptedException e) {
+      if (task != null) {
+        try {
+          task.unlock();
+        } catch (InterruptedException ignore) { // fallthrough
+        }
+      }
+      running = false;
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Set this workers running flag to false if it is not currently transferring an item
+   *
+   * @return {@code true} if the worker was cancelled, otherwise {@code false}
+   */
+  boolean cancelIfNotProcessing() {
+    if (!processing) {
+      this.running = false;
+      return true;
+    }
+    return false;
+  }
+
+  private boolean supported(Task task) {
+    if (OperationType.UNKNOWN.equals(task.getOperation())) {
+      return false;
+    }
+
+    List<MetadataInfo> infos = task.metadatas().collect(Collectors.toList());
+    if (infos.size() > 1) {
+      LOGGER.warn("More than one metadata is currently not supported by this worker");
+      return false;
+    }
+
+    DdfMetadataInfo ddfInfo =
+        infos.stream()
+            .filter(DdfMetadataInfo.class::isInstance)
+            .map(DdfMetadataInfo.class::cast)
```

Comment:
✏️ If you end up keeping this part of the method, there is no need to case and no need to find first. Just do a `count()`